src/components/editor/EditorAppController.jsx

```diff
@@ -97,7 +96,15 @@ export default class EditorAppController extends BaseComponent {
   }
 
   signOut() {
-    browserHistory.push('/login');
+    if (!window.gapi.auth2) {
```

Comment:
Also, forgot this one in my original review. Have you tested this? As I would assume that this would always be true as long as the script has loaded right? Which it does with the initial html? (though I might just not understand the API)

Reply:
I think so - gapi exists when the script has loaded.

Reply:
Yeah, so you should probably be checking just whether the script has loaded right?

Also if `window.gapi` hasn't been defined yet then `window.gapi.auth2` will throw an error since `window.gapi = undefined` so you'll be trying to do `undefined.auth2` which of course throws as it's not an object.

Reply:
I'll add the check